pom.xml

```diff
@@ -271,6 +281,221 @@
                     <artifactId>jacoco-maven-plugin</artifactId>
                     <version>${jacoco.version}</version>
                 </plugin>
+                <plugin>
+                    <groupId>org.eclipse.sisu</groupId>
+                    <artifactId>sisu-maven-plugin</artifactId>
+                    <version>${sisu.version}</version>
+                </plugin>
+                <plugin>
+                    <groupId>com.gradle</groupId>
+                    <artifactId>gradle-enterprise-maven-extension</artifactId>
+                    <configuration>
+                        <gradleEnterprise>
+                            <plugins>
+                                <plugin>
+                                    <artifactId>maven-checkstyle-plugin</artifactId>
+                                    <inputs>
+                                        <properties>
+                                            <property>
+                                                <name>propertyExpansion</name>
+                                                <value>config_loc=ignored</value>
+                                            </property>
+                                        </properties>
+                                        <fileSets>
+                                            <fileSet>
+                                                <name>checkstyleConfigs</name>
+                                                <paths>
+                                                    <path>${maven.multiModuleProjectDirectory}/config/checkstyle</path>
+                                                </paths>
+                                                <normalization>RELATIVE_PATH</normalization>
+                                            </fileSet>
+                                        </fileSets>
+                                    </inputs>
+                                </plugin>
+                                <plugin>
+                                    <groupId>com.diffplug.spotless</groupId>
+                                    <artifactId>spotless-maven-plugin</artifactId>
```

Comment:
does this add caching for both the check goals and the formatting goals?

Reply:
Well, it enables caching for both the check and the `spotless:apply` goals, however `spotless:apply` is [missing some parameters](https://ge.solutions-team.gradle.com/s/ukd66bq2tlesc/timeline?details=26o7oox6iwmxi), which I'm not sure if we want to add?

Reply:
You can enable it only per execution id. https://docs.gradle.com/enterprise/maven-extension/#declaring_additional_outputs

Given this functionality, do you think it makes sense to cache the `spotless:apply` goals?

Reply:
Hm, I was checking the [making_other_goals_cacheable](https://docs.gradle.com/enterprise/maven-extension/#making_other_goals_cacheable) part and missed that, but it did seem like a missing functionality. In light of this, I'd say it only makes sense to have the check be cached. For apply, the outputs would be the same as inputs and they aren't running apply as part of any other goal, it's on-demand only so I think it doesn't make sense to cache.

Reply:
Agreed, only the `check` should be cached. The `format` goal should not be cache because it is not idempotent.

Reply:
I don't believe so, your [link](https://github.com/gradle/develocity-build-config-samples/pull/1049/files#diff-8b4964db5582f0d23bd1a3e9438d5867fb89fbf2034eb4d80a03936742ae5456R22) only shows the spotless configuration that the `spotless:check` should be part of the general `check` goal. I believe [this](https://github.com/ribafish/micronaut-maven-plugin/blob/16d68add6f9e1b323c856ed570c55a60876779a6/pom.xml#L726-L740) is how you configure it - which I did and [this](https://ge.solutions-team.gradle.com/s/vwmzgqwrza7js/timeline?toggled=WyIzIiwiMjUiXQ&view=by-type) is the scan running both of the mojos.

Reply:
Ah you're right. Thanks for the scan link too 👍